warehouse/tuf/repository.py

```diff
@@ -0,0 +1,497 @@
+        Initialize the TUF repository from scratch, including a brand new root.
+
+        Args:
+            payload: Dictionary of the role name and the role parameters
+            store: After initialization stores the metadata files
+
+        Returns:
+             Dictionary with key as role name and value as Metadata
+             ``Dict[str, Metadata]``
+        """
+
+        self._check_is_initialized()
+
+        top_level_roles_metadata = dict()
+        if self.is_initialized:
+            raise FileExistsError("Metadata already exists in the Storage Service")
+
+        targets = Targets(1, SPEC_VERSION, payload[Targets.type].expiration, {}, None)
+        targets_metadata = Metadata(targets, {})
+        top_level_roles_metadata[Targets.type] = targets_metadata
+
+        meta = {"targets.json": MetaFile(targets.version)}
+        snapshot = Snapshot(1, SPEC_VERSION, payload[Snapshot.type].expiration, meta)
+        snapshot_metadata = Metadata(snapshot, {})
+        top_level_roles_metadata[Snapshot.type] = snapshot_metadata
+
+        snapshot_meta = MetaFile(snapshot.version)
+        timestamp = Timestamp(
+            1, SPEC_VERSION, payload[Timestamp.type].expiration, snapshot_meta
+        )
+        timestamp_metadata = Metadata(timestamp, {})
+        top_level_roles_metadata[Timestamp.type] = timestamp_metadata
+
+        roles = {
+            role_name: Role([], payload[role_name].threshold)
+            for role_name in TOP_LEVEL_ROLE_NAMES
+        }
+        root = Root(1, SPEC_VERSION, payload[Root.type].expiration, {}, roles, True)
+
+        # Sign all top level roles metadata
+        signers = dict()
+        for role in TOP_LEVEL_ROLE_NAMES:
+            if payload[role].threshold < len(payload[role].keys):
+                raise ValueError(
+                    f"Role {role} has missing Key(s) "
+                    f"to match to defined threshold {payload[role].threshold}."
+                )
+
+            for key in payload[role].keys:
+                root.add_key(role, Key.from_securesystemslib_key(key))
+
+            signers[role] = {
+                key["keyid"]: SSlibSigner(key) for key in payload[role].keys
+            }
+
+        root_metadata = Metadata(root, {})
+        top_level_roles_metadata[Root.type] = root_metadata
+        for role in signers:
+            for signer in signers[role].values():
+                top_level_roles_metadata[role].sign(signer, append=True)
+
+            if store:
+                self._store(role, top_level_roles_metadata[role])
+
+        self._is_initialized = True
+        return top_level_roles_metadata
+
+    def load_role(self, rolename: str) -> Metadata:
+        """
+        Loads a specific metadata by role name.
+
+        Args:
+            rolename: Role name
+
+        Returns:
+             Role Metadata
+             ``tuf.api.metadata.Metadata``
+        """
+        return Metadata.from_file(rolename, None, self.storage_backend)
```

Comment:
Does this miss a call to your `_filename` helper? I see you really always only pass role names as `rolename`, but `Metadata.from_file` needs a full file path as first argument.

Reply:
But it is true when using the built-in storage backend. In that case, it uses the `IStorageService` services and needs to be handled by the configured Warehouse Storage, such as SaaS.

Reply:
I added a note to the docstring that we expect the storage backend to take care of this.